### DISCUSSION TITLE
Support for Aquacal Heatwave heat pumps, specifically via PoolSync port?

### DISCUSSION THREAD

**Discussion by zbrozek** [tagyoureit/nodejs-poolController#797] [RS485]
I've tried using my Aquacal Heatwave SQ150VS via both the RS485 interface which emulates a Pentair Ultratemp, and also via Aquacal's PoolSync widget (an Espressif module in a box with an RS485 transceiver) which uses a different RS485 port on the HP9 control board. The PoolSync interface appears to allow the heat pump compressor to run at variable speed while the Ultratemp interface appears to emulate a fireman's switch.

I'd really like to be able to get variable speed control. When I was using the PoolSync my hot tub temperature was constant and didn't cycle. And the compressor was quieter, which was really nice. Is there any plan to support the Aquacal native interface? Is there anything I can do to help make that happen if not?

**Comment by rstrouse** [comment]
What are chances of you getting Aquacal to cough up the protocol.  You could hook an njsPC instance that does not have any installed equipment to the rs485 port.  There is a chance that the preamble is a known protocol.  We don’t even know what the baud rate is.  I don’t have one of these but if you wear a propeller hat and can get a byte dump we may be able to decode it.

Why do these companies do this.  In 10 years are they going to continue to support it?  There are way too many cloud apps sitting on the ash heap.

**Comment by zbrozek** [comment]
I'm guessing they're not going to provide official documentation (I've asked), so sniffing and dumping are probably the best bets. I can certainly make arbitrarily-long logs of data. I can also try reading the firmware out of the Espressif module, but I expect that's locked down.

**Reply by zbrozek** [reply]
Dumping the esp32 binary seems to have worked. I strongly suspect that all of the debug symbols are included.

**Comment by zbrozek** [comment]
OK, I've reconnected the PoolSync device to the heat pump, but put another RS485 transceiver and interface in parallel. In my case, `/dev/ttySC0` goes to the Pentair controller (running firmware 2.100) and `/dev/ttySC1` goes to the Aquacal proprietary port. Right now I haven't configured the heat pump to use the proprietary port, but I can do that at-will.

My oscilloscope tells me that the bus operates at 9600 baud and I added it as an "aux port" in nodejs-poolController. The Message Manager doesn't seem to have a raw capture mode, so I ended up disabling the port again. Here's what I get with `od`, but with the heat pump not configured to use the PoolSync controller.

```
pi@pool:/dev $ od -x < /dev/ttySC1
0000000 06fc 0001 0041 bb20 06fc 0001 0041 bb20
*
```
If I use the phone app to "discover devices" a few new messages show up:
```
0001020 06fc 0001 0041 bb20 06fc 0001 0041 bb20    # same as before
0001040 06fc 0001 0071 c97c 16fc 0001 0070 007f
0001060 0000 0000 0000 ff00 ffff ffff ffff bdff
0001100 16fc 0001 0070 007f 0000 0000 0000 ff00
0001120 ffff ffff ffff bdff 06fc 0001 0071 c97c
0001140 06fc 0001 0041 bb20 06fc 0001 0041 bb20    # same as before
```

**Reply by rstrouse** [reply]
If I were to make a guess at that I would think the preamble would be 06fc indicating the start of the packet but that doesn't mesh with the rest of the bytes.  Have you seen a ff00 ffa5 pattern in there.
